synapseutils/copy.py

```diff
@@ -199,52 +242,38 @@ def _copyFile(syn, entity, destinationId, version=None, replace=False, setProven
     #Leave this return statement for test
     return new_ent['id']
 
-#Copy Table
-def _copyTable(syn, entity, destinationId, setAnnotations=False):
+def _copyTable(syn, entity, destinationId, updateExisting=False):
     """
     Copies synapse Tables
 
     :param entity:          A synapse ID of Table Schema
 
     :param destinationId:   Synapse ID of a project that the Table wants to be copied to
 
-    :param setAnnotations:  Set the annotations of the copied table to be the annotations of the entity
-                            Defaults to False
     """
 
     print("Getting table %s" % entity)
     myTableSchema = syn.get(entity)
     #CHECK: If Table name already exists, raise value error
-    search = syn.query('select name from table where projectId == "%s"' % destinationId)
-    for i in search['results']:
-        if i['table.name'] == myTableSchema.name:
-            raise ValueError('A table named "%s" already exists in this location. Table could not be copied'%myTableSchema.name)
+    if not updateExisting:
+        existingEntity = syn._findEntityIdByNameAndParent(myTableSchema.name, parent=destinationId)
+        if existingEntity is not None:
+            raise ValueError('An entity named "%s" already exists in this location. Table could not be copied'%myTableSchema.name)
 
```

Comment:
How do you handle updates to existing tables.  It seems risky to have the updateExisting as there are so many different ways that the data could be merged.  Does it overwrite the content.  Does it append. Or does it modify rows based on the row numbers.